### ISSUE TITLE
invalid mount config for type "bind"

### ISSUE THREAD

**Issue by bradrydzewski** [drone/drone#2725]
Fix issues where bind mount fails if path does not exist on the host machine. This is related to how drone-runtime changed how it configured docker volumes when invoking the Docker API.

**Comment by matthewhartstonge** [comment]
Oh nice! wondered if it was something to do between drone/docker/moby...

For others, this was our quick workaround in the meantime:

```yaml
volumes:
  - name: tmp
    host:
      path: /tmp
  - name: cache
    host:
      path: /tmp/cache_${DRONE_COMMIT_BRANCH}/vendor

steps:
  - name: prime-cache-folder
    image: alpine:latest
    commands:
      - mkdir -p /lib/var/tmp/cache_${DRONE_COMMIT_BRANCH}/vendor
    volumes:
      - name: tmp
        path: /lib/var/tmp

- name: cache-dep-restore
    image: drillster/drone-volume-cache
    settings:
      restore: true
      mount:
        - /drone/src/github.com/drone/drone/vendor
    volumes:
      - name: cache
        path: /cache
```

**Comment by bradrydzewski** [comment]
this is fixed right now in v1.2.2 and in master but you need to use a feature flag to enable. This should be passed to the agent (or the server in single-server mode)

```
DRONE_FEATURE_VOLUME_SET=true
```

volumes are tricky because there are 4 different types (tempfs, bind mount, data volume and npipe) and I placed this behind a feature flag to reduce change of regression. I will remove the feature flag in 1.3.0 once I have confirmation that it is working as expected for people.

**Comment by moritzheiber** [comment]
Setting the feature flag solved the issue for me on 1.2.3 while using named volumes :+1:

**Comment by bradrydzewski** [comment]
removed feature flag and merged into master, available in `:latest` and will be available in 1.3 which should be tagged today or tomorrow.

**Comment by sever-sever** [comment]
How to use it with the `gitness` ?
I can't figure it out for several days.
